### PR TITLE
Add FIRST_VALUE

### PR DESCRIPTION
**Pull request by max-hoffman**

### CODE REVIEW

File: sql/expression/function/aggregation/window/first_value.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	if f.window != nil {
+		sb.WriteString(" ")
+		sb.WriteString(f.window.String())
+	}
+	return sb.String()
+}
+
+func (f *FirstValue) DebugString() string {
+	sb := strings.Builder{}
+	//sb.WriteString(fmt.Sprintf("first_value(%s)", f.Child.String()))
```
Comment (VinaiRachakonda) [inline sql/expression/function/aggregation/window/first_value.go:66]: Should be uncommented right?

File: sql/expression/function/aggregation/window/first_value.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+// WithChildren implements sql.Expression
+func (f *FirstValue) WithChildren(ctx *sql.Context, children ...sql.Expression) (sql.Expression, error) {
+	nf := *f
+	window, err := f.window.FromExpressions(children[:len(children)-1])
+	if err != nil {
+		return nil, err
+	}
+
+	nf.Child = children[len(children)-1]
+	nf.window = window
```
Comment (VinaiRachakonda) [inline sql/expression/function/aggregation/window/first_value.go:111]: You should just use WithWindow()

File: sql/expression/function/aggregation/window/first_value.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+func (f *FirstValue) Children() []sql.Expression {
+	if f == nil {
+		return nil
+	}
+	return append(f.window.ToExpressions(), f.Child)
+}
+
+// WithChildren implements sql.Expression
+func (f *FirstValue) WithChildren(ctx *sql.Context, children ...sql.Expression) (sql.Expression, error) {
+	nf := *f
```
Comment (VinaiRachakonda) [inline sql/expression/function/aggregation/window/first_value.go:104]: Should be an error check to make sure there are least children

File: enginetest/enginetests.go [hunk 4]
```diff
@@ -2395,6 +2395,33 @@ func TestWindowAgg(t *testing.T, harness Harness) {
...
+	TestQuery(t, harness, e, `SELECT a, first_value(a) over (partition by b) FROM t1 order by a`, []sql.Row{
+		{0, 0},
+		{1, 1},
+		{2, 2},
+		{3, 0},
+		{4, 1},
+		{5, 5},
+	}, nil, nil)
+
+	TestQuery(t, harness, e, `SELECT a, first_value(a-1) over (partition by b order by c) FROM t1 order by a`, []sql.Row{
```
Comment (VinaiRachakonda) [inline enginetest/enginetests.go:2417]: Can you add a test case that does a first value over another column that is not a?

**Review by VinaiRachakonda — APPROVED** [review]
LGTM! Some more tests would be nice

File: sql/expression/function/aggregation/window/first_value.go (hunk 2 above)
Comment (max-hoffman) [reply sql/expression/function/aggregation/window/first_value.go:111]: i get the point of the helper function if I didn't have a child expr, but isn't it a bit awkward to convert back to a pointer and handle an error just to call a function that sets window?

File: sql/expression/function/aggregation/window/first_value.go (hunk 3 above)
Comment (max-hoffman) [reply sql/expression/function/aggregation/window/first_value.go:104]: error if children < 2? required arg expression and at least one window expression?

File: enginetest/enginetests.go (hunk 4 above)
Comment (max-hoffman) [reply enginetest/enginetests.go:2417]: 👍

File: sql/expression/function/aggregation/window/first_value.go (hunk 3 above)
Comment (VinaiRachakonda) [reply sql/expression/function/aggregation/window/first_value.go:108]: Yes 2 my bad. Typo on my part.

**Review by zachmu — APPROVED** [review]
LGTM, thanks for jumping on this

**Pull request merged**